src/main/scala/zio/rocksdb/transaction/package.scala

```diff
@@ -0,0 +1,41 @@
+package zio.rocksdb
+
+import org.{ rocksdb => jrocks }
+import zio._
+
+package object transaction {
+  type Bytes         = Array[Byte]
+  type Transaction   = Has[RocksDB.TransactionService]
+  type TransactionDB = Has[RocksDB.TransactionDBService]
+
+  object RocksDB {
+    trait TransactionService {
+      def get(readOptions: jrocks.ReadOptions, key: Bytes): Task[Option[Bytes]]
+      def getForUpdate(readOptions: jrocks.ReadOptions, key: Bytes, exclusive: Boolean): Task[Option[Bytes]]
+      def put(key: Bytes, value: Bytes): Task[Unit]
+      def delete(key: Bytes): Task[Unit]
+      def commit: Task[Unit]
+    }
+
+    trait TransactionDBService {
+      def beginTransaction(writeOptions: jrocks.WriteOptions): UIO[TransactionService]
+    }
+
+    def transaction[R <: Has[_], E >: Throwable, A](
```

Comment:
Let’s move this out of the `object RocksDB` and into the package object, and call it `transactionally`.

Reply:
Actually, a more concise option would be `atomically`.

Reply:
Done.